krummstab/__init__.py

```diff
@@ -827,10 +863,11 @@ def create_individual_marks_file(_the_config: config.Config) -> None:
     individual_marks.update({"adam_sheet_name": get_adam_sheet_name_string()})
     if _the_config.points_per == "exercise" and _the_config.marking_mode == "exercise":
         individual_marks.update({"exercises": args.exercises})
-    for team_dir, mark in marks.items():
-        for first_name, last_name, email in args.team_dir_to_team[team_dir]:
+    for team_dir in get_relevant_team_dirs():
+        submission_info = get_submission_info(team_dir)
+        for first_name, last_name, email in submission_info.get("team"):
             key = (f"{email}".lower())
-            marks_dict.update({key: mark})
+            marks_dict.update({key: marks.get(get_team_key(submission_info))})
```

Comment:
The variable names are a bit confusing here. I would rename
* `individual_marks` -> `file_content`
* `key` -> `student_key`
* `get_team_key(submission_info)` -> `team_key` (new local variable, defined before the `for` loop
* `marks_dict` -> `team_marks`
* `marks` -> `student_marks`

Since we're at this, `f"{email}".lower()` is the same as `email.lower()` and `d.update({"key": value})` is the same as `d["key"] = value`. But instead of simplifying all `update`s, I would define `file_content` in one go, just before the `with open(...)` line:

```Python
file_content = {
    "tutor_name": _the_config.tutor_name,
    "adam_sheet_name": get_adam_sheet_name_string(),
    "marks": marks_dict
}
if _the_config.points_per == "exercise" and _the_config.marking_mode == "exercise":
        file_content["exercises"] = args.exercises
```